tests/Unit/Parallel/PhaseControl/Test_CheckpointAndExitAfterWallclock.cpp

```diff
@@ -0,0 +1,145 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <optional>
+#include <utility>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "Framework/TestCreation.hpp"
+#include "Options/Protocols/FactoryCreation.hpp"
+#include "Parallel/GlobalCache.hpp"
+#include "Parallel/PhaseControl/CheckpointAndExitAfterWallclock.hpp"
+#include "Parallel/PhaseControl/PhaseControlTags.hpp"
+#include "ParallelAlgorithms/EventsAndTriggers/LogicalTriggers.hpp"
+#include "ParallelAlgorithms/EventsAndTriggers/Trigger.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/ProtocolHelpers.hpp"
+#include "Utilities/TMPL.hpp"
+
+struct Metavariables {
+  using component_list = tmpl::list<>;
+
+  struct factory_creation
+      : tt::ConformsTo<Options::protocols::FactoryCreation> {
+    using factory_classes =
+        tmpl::map<tmpl::pair<Trigger, tmpl::list<Triggers::Always>>>;
+  };
+
+  enum class Phase { PhaseA, WriteCheckpoint, Exit };
+};
+
+SPECTRE_TEST_CASE("Unit.Parallel.PhaseControl.CheckpointAndExitAfterWallclock",
+                  "[Unit][Parallel]") {
+  // note that the `contribute_phase_data_impl` function is currently untested
+  // in this unit test, because we do not have good support for reductions in
+  // the action testing framework. These are tested in the integration test
```

Comment:
In a strict sense, the `contribute_phase_data_impl` for _this_ class isn't tested in the `AlgorithmPhaseControl.cpp`, so maybe remove or rephrase the last sentence

Reply:
removed